lib/ansible/module_utils/basic.py

```diff
@@ -1949,15 +1948,16 @@ def _return_formatted(self, kwargs):
                     self.warn(w)
             else:
                 self.warn(kwargs['warnings'])
+
         if self._warnings:
             kwargs['warnings'] = self._warnings
 
         if 'deprecations' in kwargs:
-            if isinstance(kwargs['deprecations'], list):
-                for d in kwargs['deprecations']:
-                    self.warn(d)
-            else:
-                self.warn(kwargs['deprecations'])
+            for d in kwargs['deprecations']:
+                if isinstance(d, SEQUENCETYPE) and len(d) == 2:
```

Comment:
`deprecations` doesn't seem used by any module, the version could be always required (even if empty/None).